src/record/index.js

```diff
@@ -7,25 +7,31 @@ const errcode = require('err-code')
  */
 class Record {
```

Comment:
This is extended in https://github.com/libp2p/js-libp2p/tree/master/src/record/peer-record . We can also have a types definition, but it seems redundant. What do you think?

Reply:
If thing does not need to be a class it's best to define it as interface / type because it could be substituted with other implementation or even just plain object that conforms the interface. Classes should really just be a one possible implementation of it. Current version implies that TS will error anywhere `record instanceof Record` isn't true.